Source/ImPlotWidgets/Private/LineFilledGraphWidget.cpp

```diff
@@ -66,22 +72,28 @@ void SFilledGraphWidget::OnImGuiDraw() const
 		ImPlot::PushStyleVar(ImPlotStyleVar_PlotPadding, ImVec2(20, 20));
 		ImPlot::PushStyleVar(ImPlotStyleVar_LabelPadding, ImVec2(10, 10));
 
-		ImPlot::SetNextPlotLimits(0, XMaxValue, YMinValue*1.1f, YMaxValue*1.1f, ImGuiCond_Always);
-		if (ImPlot::BeginPlot(TCHAR_TO_ANSI(*Title), TCHAR_TO_ANSI(*XLabel), TCHAR_TO_ANSI(*YLabel), ImVec2(-1,-1), GraphFlags, AxisFlags, AxisFlags))
+		ImPlot::SetNextPlotLimits(0, XMaxValue, FUnitConversion::Convert(YMinValue, YUnit, BestFitUnit)*1.1f, FUnitConversion::Convert(YMaxValue, YUnit, BestFitUnit) * 1.1f, ImGuiCond_Always);
+		if (ImPlot::BeginPlot(TCHAR_TO_ANSI(*Title), TCHAR_TO_ANSI(*XLabel), TCHAR_TO_ANSI(*(YLabel + " (" + FUnitConversion::GetUnitDisplayString(BestFitUnit) + ")" )), ImVec2(-1,-1), GraphFlags, AxisFlags, AxisFlags))
 		{
 			for (int32 CurrentArray = 0; CurrentArray < XValueArrays.Num(); ++CurrentArray)
 			{
 				if (ShowLine[CurrentArray])
 				{
-					ImPlot::PushStyleVar(ImPlotStyleVar_FillAlpha, 0.25f);
-					ImPlot::PlotShaded(TCHAR_TO_ANSI(*DataSets[CurrentArray].Name), XValueArrays[CurrentArray].GetData(), YValueArrays[CurrentArray].GetData(), YValueArrays[CurrentArray].Num(), 0);
-					ImPlot::PopStyleVar();
-
-					ImPlot::PushStyleVar(ImPlotStyleVar_LineWeight, 2.f);
-					ImPlot::PlotLine(TCHAR_TO_ANSI(*DataSets[CurrentArray].Name), XValueArrays[CurrentArray].GetData(), YValueArrays[CurrentArray].GetData(), YValueArrays[CurrentArray].Num(), 0);
-					ImPlot::PopStyleVar();
+					if (FittedYValues.Num() > 0)
+					{
+						VisibleLineMaxValues.Add(YMaxValues[CurrentArray]);
+
+						ImPlot::PushStyleVar(ImPlotStyleVar_LineWeight, 2.f);
+						ImPlot::PlotLine(TCHAR_TO_ANSI(*DataSets[CurrentArray].Name), XValueArrays[CurrentArray].GetData(), FittedYValues[CurrentArray].GetData(), FittedYValues[CurrentArray].Num(), 0);
+						ImPlot::PopStyleVar();
+					}
 				}
+
 			}
+
+			YMaxValue = FMath::Max(MoveTemp(VisibleLineMaxValues));
+			BestFitUnit = FUnitConversion::QuantizeUnitsToBestFit(YMaxValue, YUnit).Units;
```

Comment:
WRT (this comment) I would place these inside the `RecalculateFittedLines()` too